Source: arch, https://git.archlinux.org/svntogit/packages.git/plain/trunk/765567_non-recursive_string_subst.patch?h=packages/docbook-xsl
        also in debian, backport of git master
Upstream: Fixed in git master, https://github.com/docbook/xslt10-stylesheets/commit/1c41ae5ae3d4816687467712580d5f05957b3b88

Description: use EXSLT "replace" function when available
 A recursive implementation  of string.subst is problematic,
 long strings with many matches will cause stack overflows.
Author: Peter De Wachter <pdewacht@gmail.com>
Bug-Debian: https://bugs.debian.org/750593

diff -Naur docbook-xsl-nons-1.79.2/lib/lib.xsl docbook-xsl-nons-1.79.2.new/lib/lib.xsl
--- docbook-xsl-nons-1.79.2/lib/lib.xsl	2016-12-09 23:47:31.000000000 +0100
+++ docbook-xsl-nons-1.79.2.new/lib/lib.xsl	2018-07-20 14:53:49.912030595 +0200
@@ -6,7 +6,10 @@
 
      This module implements DTD-independent functions
 
-     ******************************************************************** --><xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
+     ******************************************************************** --><xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:str="http://exslt.org/strings"
+                exclude-result-prefixes="str"
+                version="1.0">
 
 <xsl:template name="dot.count">
   <!-- Returns the number of "." characters in a string -->
@@ -52,6 +55,9 @@
   <xsl:param name="replacement"/>
 
   <xsl:choose>
+    <xsl:when test="function-available('str:replace')">
+      <xsl:value-of select="str:replace($string, string($target), string($replacement))"/>
+    </xsl:when>
     <xsl:when test="contains($string, $target)">
       <xsl:variable name="rest">
         <xsl:call-template name="string.subst">
